test/tests/functional/pbs_svr_dyn_res.py

```diff
@@ -480,3 +480,237 @@ def test_res_size_runtime(self):
         job_comment += " foobar (R: 95gb A: 50gb T: 50gb)"
         a = {'job_state': 'Q', 'comment': job_comment}
         self.server.expect(JOB, a, id=jid, attrop=PTL_AND)
+
+    def test_res_size_invalid_input(self):
+        """
+        Test invalid values returned from server_dyn_resource
+        script for resource type 'size'.
+        Script returns a 'string' instead of type 'size'.
+        """
+        # Create a resource of type size
+        resname = ["foobar"]
+        restype = ["size"]
+        resflag = ["q"]
+
+        # Script returns invalid value for resource type 'size'
+        script_body = "echo two gb"
+
+        fn = self.du.create_temp_file(prefix="PtlPbs_size",
+                                      suffix=".scr",
+                                      body=script_body)
+        self.du.chmod(path=fn, mode=0755, sudo=True)
+
+        resval = ['"' + resname[0] + ' ' + '!' + fn + '"']
+
+        self.setup_dyn_res(resname, restype, resval, resflag)
+
+        # Submit job
+        a = {'Resource_List.foobar': '2gb'}
+        j = Job(TEST_USER, attrs=a)
+        jid = self.server.submit(j)
+
+        # The job shouldn't run
+        job_comment = "Can Never Run: Insufficient amount of server resource:"
+        job_comment += " foobar (R: 2gb A: 0kb T: 0kb)"
+        a = {'job_state': 'Q', 'comment': job_comment}
+        self.server.expect(JOB, a, id=jid, attrop=PTL_AND)
+
+    def test_res_string_invalid_input1(self):
+        """
+        Test invalid values returned from server_dyn_resource
+        script for resource type 'string'
+        Script returns a 'integer' instead of type 'string'.
```

Comment:
I dislike this test.  You aren't testing what you are suggesting.  The server_dyn_res script does return a string.  It returns "100" (not the number, but the string "100").  Nothing is really going wrong here.  The job is not running because your strings do not match.  I say delete this test.

Reply:
Yes. Realized late that couple of these tests are not adding any value as the script is always adding a string type. I am deleting those tests.